.github/workflows/ccpp.yml

```
@@ -37,7 +37,9 @@ jobs:
                   build-essential \
                   gcc-multilib \
                   g++-multilib \
-                  dejagnu
+                  dejagnu;
+          # install Rust directly using rustup
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y;
```

I think we do want to pin the versions of rustup and rustc. `sh.rustup.rs` does not provides a way to fetch an older version of rustup but the resulting script provides a way to select a given toolchain version, I guess that's better than nothing.

```suggestion
          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain=1.49.0;
```

---

Here is the url for a pinned rustup version:

```
https://static.rust-lang.org/rustup/archive/1.27.0/x86_64-unknown-linux-gnu/rustup-init
```